base/compiler/ssair/inlining.jl

```diff
@@ -994,6 +1024,22 @@ function inline_invoke!(ir::IRCode, idx::Int, sig::Signature, invoke_data::Invok
     return nothing
 end
 
+function narrow_opaque_closure!(ir::IRCode, stmt::Expr, @nospecialize(calltype))
+    if isa(calltype, PartialOpaque)
+        lbt = argextype(stmt.args[2], ir, ir.sptypes)
+        lb, exact = instanceof_tfunc(lbt)
+        exact || return
+        # Narrow opaque closure type
+        if isa(calltype.ci, OptimizationState)
```

Comment:
```suggestion
        source = calltype.ci
        if isa(source, OptimizationState)
        ...
```
(same nit: avoiding recomputing values)